modules/friendlytalkback.js

```diff
@@ -358,55 +378,15 @@ var callback_evaluate = function(e) {
 	var talkpage = new Morebits.wiki.page(fullUserTalkPageName, 'Adding talkback');
 	var tbPageName = tbtarget === 'mytalk' ? mw.config.get('wgUserName') : page;
 
-	var text;
+	var text = '\n\n';
 	if (tbtarget === 'notice') {
-		switch (page) {
-			case 'afchd':
-				text = '\n\n{{subst:AFCHD/u|' + section + '}} ~~~~';
-				talkpage.setEditSummary('You have replies at the [[Wikipedia:AFCHD|Articles for Creation Help Desk]]' + Twinkle.getPref('summaryAd'));
-				break;
-			case 'an':
-				text = '\n\n== ' + Twinkle.getFriendlyPref('adminNoticeHeading') + ' ==\n';
-				text += '{{subst:ANI-notice|thread=' + section + "|noticeboard=Wikipedia:Administrators' noticeboard}} ~~~~";
-				talkpage.setEditSummary("Notice of discussion at [[Wikipedia:Administrators' noticeboard]]" + Twinkle.getPref('summaryAd'));
-				break;
-			case 'an3':
-				text = '\n\n{{subst:An3-notice|' + section + '}} ~~~~';
-				talkpage.setEditSummary("Notice of discussion at [[Wikipedia:Administrators' noticeboard/Edit warring]]" + Twinkle.getPref('summaryAd'));
-				break;
-			case 'ani':
-				text = '\n\n== ' + Twinkle.getFriendlyPref('adminNoticeHeading') + ' ==\n';
-				text += '{{subst:ANI-notice|thread=' + section + "|noticeboard=Wikipedia:Administrators' noticeboard/Incidents}} ~~~~";
-				talkpage.setEditSummary("Notice of discussion at [[Wikipedia:Administrators' noticeboard/Incidents]]" + Twinkle.getPref('summaryAd'));
-				break;
-			case 'coin':
-				text = '\n\n{{subst:Coin-notice|thread=' + section + '}} ~~~~';
-				talkpage.setEditSummary('Notice of discussion at [[Wikipedia:Conflict of interest noticeboard]]' + Twinkle.getPref('summaryAd'));
-				break;
-			case 'drn':
-				text = '\n\n{{subst:DRN-notice|thread=' + section + '}} ~~~~';
-				talkpage.setEditSummary('Notice of discussion at [[Wikipedia:Dispute resolution noticeboard]]' + Twinkle.getPref('summaryAd'));
-				break;
-			case 'hd':
-				text = '\n\n== Your question at the Help desk ==\n';
-				text += '{{helpdeskreply|1=' + section + '|ts=~~~~~}}';
-				talkpage.setEditSummary('You have replies at the [[Wikipedia:Help desk|Wikipedia help desk]]' + Twinkle.getPref('summaryAd'));
-				break;
-			case 'otrs':
-				text = '\n\n{{OTRSreply|1=' + section + '|2=~~~~}}';
-				talkpage.setEditSummary('You have replies at the [[Wikipedia:OTRS noticeboard|OTRS noticeboard]]' + Twinkle.getPref('summaryAd'));
-				break;
-			case 'th':
-				text = "\n\n== Teahouse talkback: you've got messages! ==\n{{WP:Teahouse/Teahouse talkback|WP:Teahouse/Questions|" + section + '|ts=~~~~}}';
-				talkpage.setEditSummary('You have replies at the [[Wikipedia:Teahouse/Questions|Teahouse question board]]' + Twinkle.getPref('summaryAd'));
-				break;
-			default:
-				throw 'Twinkle.talkback, function callback_evaluate: default case reached';
-		}
+		text += Morebits.string.safeReplace(Twinkle.talkback.noticeboards[page].text, '$SECTION', section);
```

Comment:
Just to account for the very rare scenario where section includes a `$&` or `$'` or $` in it, which would cause JS replace() to give funny results.